File: constants/nativeStabilizationFlags.ts
/**
 * Mirrors Java {@code NativeStabilizationFlags} for JS intervention routing.
 * Native FGS posts /events and shows system notifications when backgrounded; JS still handles foreground.
 */
export const NativeStabilizationFlags = {
  /** When true, foreground `notification` tier may use expo-notifications; native handles background. */
  ENABLE_NATIVE_POST_EVENT_INTERVENTIONS: true,
  /** Native floating Spike overlay when permitted; in-app FocusEnforcementModal unchanged. */
  ENABLE_FOCUS_GUARD_OVERLAY: true,
} as const;

File: android/app/src/main/java/com/spikesense/NativeStabilizationFlags.java
package com.spikesense;

/**
 * Native feature flags for post-/events interventions (kept in sync with
 * {@code constants/nativeStabilizationFlags.ts} conceptually).
 */
public final class NativeStabilizationFlags {

    private NativeStabilizationFlags() {
    }

    /** When true, native code may post notifications / overlay after FGS event responses. */
    public static final boolean ENABLE_NATIVE_POST_EVENT_INTERVENTIONS = true;

    /** When true, native floating Spike overlay may be used (requires overlay permission). */
    public static final boolean ENABLE_FOCUS_GUARD_OVERLAY = true;
}
